### PR TITLE
feat: full orchestration of the variant index dag

### PR DESCRIPTION
**Pull request by ireneisdoomed**
## ✨ Context
This PR describes changes in the business logic and the orchestration part to generate a full variant index starting with the defined inputs:
- Parquet files with credible sets generated by Gentropy
- Uniprot evidence file (pre ETL)
- PharmGKB evidence file (pre ETL)
- ClinVar evidence file (pre ETL)

and continuing to their annotation with VEP (more [here](https://github.com/opentargets/gentropy/pull/641))

## 🛠 What does this PR implement

- Generation of VCF files is done with Airflow. The sources to be included are configured with a YAML
- Annotation of variant coordinates from RSIDs with Ensembl API calls
- Implementation of a new step that takes one dataset containing variant info as input, and produces a valid VCF file
- Rearrangement of the tasks that annotate VCFs and produce a variant index from the VCF output into the variant index DAG
- Redefinition of variant ID: `ot_var_{hashed variant id}` for variants longer than 100 characters, `variant ID` for shorter variants as discussed with @d0choa and @DSuveges 

## 🙈 Missing

**There is a problem in the XCOM between the task that generates the list of files and the task that runs VEP. Listing the location of the VCFs is not returning anything and the annotation fails with `No files found to process.`**

@DSuveges , can you help me with this? It must be a tiny issue that I am not able to capture. I've tested the VCF files generation with a subset of data and it runs, and the same goes for the VEP annotation, so it's only this link between the 2 that is failing.

In the meantime I am going to start porting this logic to the orchestration layer.

## 🚦 Before submitting

- [ ] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

**Review by DSuveges — CHANGES_REQUESTED** [review]
OK, we can get back to this PR later.

File: src/gentropy/dataset/variant_index.py [hunk 1]
```diff
@@ -34,6 +35,123 @@ def get_schema(cls: type[VariantIndex]) -> StructType:
...
+
+            except Exception as e:
+                raise Exception(
+                    f"Error processing batch {i // batch_size + 1}: {e}"
+                ) from e
+
+            time.sleep(pause_time)
+
+        return all_results
+
```
[the remark this answers is not in the document]
Comment (ireneisdoomed) [reply src/gentropy/dataset/variant_index.py:154]: I think that this method is useful when you're operating with a variant index. I can move the logic to the ensembl datasource, and add a method in the variant index that calls it. What do you think?

File: src/gentropy/dataset/variant_index.py [hunk 2]
```diff
@@ -34,6 +35,123 @@ def get_schema(cls: type[VariantIndex]) -> StructType:
         """
         return parse_spark_schema("variant_index.json")
 
+    @classmethod
+    def assign_variant_id(
```
[the remark this answers is not in the document]
Comment (ireneisdoomed) [reply src/gentropy/dataset/variant_index.py:39]: Is the implementation different?

[the remark this answers is not in the document]
Comment (DSuveges) [reply src/gentropy/dataset/variant_index.py:39]: [Somewhat](https://github.com/opentargets/gentropy/blob/cb7c384a28cb886b339c9e58494d1fd7c6bdafed/src/gentropy/dataset/variant_index.py#L63).

File: src/gentropy/dataset/variant_index.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DSuveges) [reply src/gentropy/dataset/variant_index.py:154]: > I can move the logic to the ensembl datasource, and add a method in the variant index that calls it.
Why would a method in variant index call that method? If you are operating with variant index, you already have the coordinates. I can't see the usecase. Even if we were building a variant index based on a list of rsids.:

```python
vi = variant_index_from_rsids(rsid_list)
```

This logic should be living somewhere else. Just the returned value would be converted to vi dataset. To me, any method that does anything with API calls should live in a very special and isolated place.

**Review by DSuveges — APPROVED** [review]
There are some caveats of this implementation, but it works. Merging would unblock some variant index related work.

**Pull request merged**